### DISCUSSION TITLE
New User interface for Stock-Patterns - Feedback needed

### DISCUSSION THREAD

**Discussion by BennyThadikaran** [BennyThadikaran/stock-pattern#87] [Announcements]
Since the inclusion of new patterns the UI has become quite cluttered. I am working on a new UI for Stock-Patterns.

My goal was to minimise the number of keystrokes required to select the desired pattern while providing a clean and clear interface.

![ui-1](https://github.com/user-attachments/assets/1fcc1577-d722-42f0-9ddf-180800410443)

![ui-2](https://github.com/user-attachments/assets/46cf32c8-3735-4d6d-8714-9982bc0e0563)

If you like to give it a try:

1. Download [test.txt](https://github.com/user-attachments/files/17827676/test.txt) to your project /src folder 
2. Rename to test.py and run it.

This only has the user interface component and nothing else. 

- You can use arrow keys or `j` and `k` to select options.
- You can also type a number to select the relevant option. 

Do provide your feedback and comments. It will be included in the next update with some fixes to existing Harmonic patterns.

**Comment by keltik85** [comment]
For me the user interface is not so relevant, what i would like to have is that passing the `user.json` config parameters using the command line like so, for example:

```
python init.py --data-path="some-x/path-y" --loader="some-loader-a" --date-column="some-date-col-label-z" --twenty-four-seven
```

**Reply by BennyThadikaran** [reply]
You have a good point. I have been thinking about exposing some options in the CLI. For example the number of candles loaded for analysis. 

I do want to keep a balance. Since too many CLI options will be overwhelming for the user, obscuring the more important options. 

Some of what you mentioned can be solved with having multiple configs. For example a config specific for Crypto and another for the stock market. 

`py init.py -c binance.json`

`py init.py -c nyse.json`

But I do agree more options must be exposed in the CLI and the configuration to be useful. I'll consider this in a future update.

Thanks for the input :+1:

**Reply by keltik85** [reply]
i can also try to open a pull request, but at the moment i was ust trying to figure something else out with your library. :-)
